File: SSP.PureInsuranceRestAPIHandler/SSP.PureInsuranceRestAPIHandler/BaseClasses/BaseQuoteRiskMsgType.cs
using SSP.PureInsuranceRestAPIHandler.Enums;
using System.Collections.Generic;

namespace SSP.PureInsuranceRestAPIHandler.BaseClasses
{
    public class BaseQuoteRiskMsgType : BaseQuoteType
    {
        public string AccountExecutiveShortname { get; set; }
        public string AccountHandlerShortname { get; set; }
        public string AlternateReference { get; set; }
        public double AmountToFinance { get; set; }
        public bool AmountToFinanceSpecified { get; set; }
        public string BankAccountName { get; set; }
        public string BankAccountNo { get; set; }
        public BaseAddressType BankAddress { get; set; }
        public string BankAreaCode { get; set; }
        public string BankBranch { get; set; }
        public string BankExtn { get; set; }
        public string BankFax { get; set; }
        public string BankFaxCode { get; set; }
        public string BankName { get; set; }
        public string BankPhone { get; set; }
        public string BankSortCode { get; set; }
        public System.Collections.Generic.List<BaseQuoteRiskMsgTypeClaim> Claim { get; set; }
        public string CoInsurancePlacement { get; set; }
        public int DayOfWeekOrMonth { get; set; }
        public bool DayOfWeekOrMonthSpecified { get; set; }
        public int FinanceCompanyNo { get; set; }
        public bool FinanceCompanyNoSpecified { get; set; }
        public System.DateTime FinanceEndDate { get; set; }
        public bool FinanceEndDateSpecified { get; set; }
        public System.DateTime FinancePreferredDate { get; set; }
        public bool FinancePreferredDateSpecified { get; set; }
        public System.DateTime FinanceQuoteDate { get; set; }
        public bool FinanceQuoteDateSpecified { get; set; }
        public int FinanceSchemeNo { get; set; }
        public bool FinanceSchemeNoSpecified { get; set; }
        public int FinanceSchemeVersion { get; set; }
        public bool FinanceSchemeVersionSpecified { get; set; }
        public System.DateTime FinanceStartDate { get; set; }
        public bool FinanceStartDateSpecified { get; set; }
        public System.DateTime InceptionDate { get; set; }
        public bool InceptionDateSpecified { get; set; }
        public System.DateTime InceptionDateTPI { get; set; }
        public bool InceptionDateTPISpecified { get; set; }
        public System.DateTime LapsedDate { get; set; }
        public bool LapsedDateSpecified { get; set; }
        public string LapsedReasonCode { get; set; }
        public string LapsedReasonDescription { get; set; }
        public string OldPolicyNumber { get; set; }
        public bool PaymentProtection { get; set; }
        public bool PaymentProtectionSpecified { get; set; }
        public string PolicyStatusCode { get; set; }
        public int PolicyVersion { get; set; }
        public string PolicyVersionTypeCode { get; set; }
        public System.DateTime RenewalDate { get; set; }
        public bool RenewalDateSpecified { get; set; }
        public BaseTaxesType[] Taxes { get; set; }
        public System.Collections.Generic.List<BaseQuoteRiskMsgTypeRisks> Risks { get; set; }
        public int SAMStagingPolicyKey { get; set; }
        public PolicyProcessTypes PolicyProcessType { get; set; }
        public string TransactionTypeCode { get; set; }
        public string NewQuoteRef { get; set; }
        public double CommissionRate { get; set; }
        public double CommissionValue { get; set; }
        public System.DateTime TransactionDueDate { get; set; }
        public string LastTransDescription { get; set; }
        public string BusinessTypeCode { get; set; }
        public string MTAReasonCode { get; set; }
        public bool IsMarketPlacePolicy { get; set; }
        public double OverrideNetPremium { get; set; }
        public int PartyKey { get; set; }
        public int SourceId { get; set; }
        public bool SkipGenerateRenewalPolicyNumber { get; set; }
        public int InsuranceFolderKey { get; set; }
        public int DeletePolicyUnderRenewal { get; set; }
        public string RenewalStatusCode { get; set; }
        public bool DoNotCopyRiskAtRenSelection { get; set; }
        public bool IsBDXRequest { get; set; }
        public System.DateTime AnniversaryDate { get; set; }
        public bool AnniversaryDateSpecified { get; set; }
        public string CollectionFrequencyCode { get; set; }
        public string PaymentTermCode { get; set; }
        public System.Collections.Generic.List<BaseUpdateCoinsuranceValuesRequestTypeRow> CoInsurers { get; set; }
    }
}

File: Orion/Components/ImportExport/Interface/SiriusImport/RestContracts/BaseQuoteRiskMsgType.vb
Public Class BaseQuoteRiskMsgType
    ' From BaseQuoteType
    Public Property AgentKey As Integer
    Public Property AlternativeRef As String
    Public Property AnalysisCode As String
    Public Property CoverEndDate As Date
    Public Property CoverStartDate As Date
    Public Property CurrencyCode As String
    Public Property Description As String
    Public Property InsuredName As String
    Public Property ProductCode As String
    Public Property QuoteRef As String
    Public Property UnderwritingYearCode As String
    ' BaseQuoteRiskMsgType additions
    Public Property AccountExecutiveShortname As String
    Public Property AccountHandlerShortname As String
    Public Property AlternateReference As String
    Public Property BranchCode As String
    Public Property BusinessTypeCode As String
    Public Property CoInsurancePlacement As String
    Public Property CoInsurers As List(Of BaseUpdateCoinsuranceValuesRequestTypeRow)
    Public Property CollectionFrequencyCode As String
    Public Property CommissionRate As Double
    Public Property CommissionValue As Double
    Public Property DeletePolicyUnderRenewal As Integer
    Public Property DoNotCopyRiskAtRenSelection As Boolean
    Public Property IsBDXRequest As Boolean
    Public Property LastTransDescription As String
    Public Property MTAReasonCode As String
    Public Property NewQuoteRef As String
    Public Property OldPolicyNumber As String
    Public Property PartyKey As Integer
    Public Property PaymentTermCode As String
    Public Property PolicyProcessType As PolicyProcessTypes?
    Public Property PolicyStatusCode As String
    Public Property Risks As List(Of BaseQuoteRiskMsgTypeRisks)
    Public Property SAMStagingPolicyKey As Integer
    Public Property SkipGenerateRenewalPolicyNumber As Boolean
    Public Property Taxes As List(Of BaseTaxesType)
    Public Property TransactionDueDate As Date
    Public Property TransactionTypeCode As String
End Class
